### PR TITLE
adding pylibmc

### PR DESCRIPTION
**Pull request by bollwyvl**
> blocked by [libmemcached](https://github.com/conda-forge/staged-recipes/pull/760)

Adds pylibmc.

### CODE REVIEW

**Comment by bollwyvl** [comment]
Hm, circle fails on:
```
gcc -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -m -m -m64 -fPIC -DUSE_ZLIB -I/opt/conda/envs/_build/include -I/opt/conda/envs/_build/include/python3.4m -c src/_pylibmcmodule.c -o build/temp.linux-x86_64-3.4/src/_pylibmcmodule.o -fno-strict-aliasing -std=c99
gcc: error: unrecognized command line option ‘-m’
```

It looks like those two `-m`s are getting created by `toolchain`:
```bash
# for py27
+++ export 'CFLAGS= -m64'
+++ CFLAGS=' -m64'
```
and even though:
```bash
++++ unset CFLAGS
```
then later:
```bash
# for py34
+++ export 'CFLAGS=-m -m'
+++ CFLAGS='-m -m'
```

Trying with an explicit `deactivate`...

**Comment by jakirkham** [comment]
Yeah, I have seen this too. I believe there is some funkiness with `conda-build`, but I forget the exact solution. It may have been a downgrade. 😒

@msarahan, bringing this one to your attention.

**Comment by jakirkham** [comment]
Yep, same story as this thread. ( https://github.com/conda-forge/staged-recipes/pull/751 )

**Comment by jakirkham** [comment]
Could you try something really silly for me @bollwyvl? Could you please try stuffing two spaces in all the environment variables from the toolchain?

**Comment by bollwyvl** [comment]
> Could you please try stuffing two spaces in all the environment variables from the toolchain?

Before or after activate?

**Comment by jakirkham** [comment]
After activate.

**Comment by jakirkham** [comment]
If this works, we should add this to the `toolchain` as a workaround and file a `conda-build` issue (before I forget again 😜).

**Comment by bollwyvl** [comment]
So, like:
```shell
source activate "${CONDA_DEFAULT_ENV}"
export CFLAGS="  "
export CXXFLAGS="  "
export LDFLAGS="  "
export LINKFLAGS="  "
```

**Comment by jakirkham** [comment]
Like...

```shell
source activate "${CONDA_DEFAULT_ENV}"
export CFLAGS="${CFLAGS}  "
export CXXFLAGS="${CXXFLAGS}  "
export LDFLAGS="${LDFLAGS}  "
export LINKFLAGS="${LINKFLAGS}  "
```

**Comment by bollwyvl** [comment]
tried that, same old same old on my local docker run...

**Comment by jakirkham** [comment]
@msarahan is looking into a fix for the chopped up env vars. Maybe we should put a hold on this so he can have some time to work on it. Sound good? 😄

**Comment by bollwyvl** [comment]
Oh indeed! Thanks @msarahan!

**Comment by pelson** [comment]
I restarted Circle and it is passing. I guess you can probably remove the activate now too (though I have been out of the loop for a few weeks, so not 100% sure)

**Comment by jakirkham** [comment]
Not quite. While `conda-build` has adopted this behavior, we still are pinned to version that do not have it in some cases.

**Comment by pelson** [comment]
So, suggest we merge and then put a reminder in https://github.com/conda-forge/conda-smithy/issues/222 to update the feedstock.

File: recipes/pylibmc/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+source deactivate
```
Comment (jakirkham) [inline recipes/pylibmc/build.sh]: This can be dropped.

File: recipes/pylibmc/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+source activate "${CONDA_DEFAULT_ENV}"
```
Comment (jakirkham) [inline recipes/pylibmc/build.sh]: Please add this comment above.

```
# FIXME: This is a hack to make sure the environment is activated.
# The reason this is required is due to the conda-build issue
# mentioned below.
#
# https://github.com/conda/conda-build/issues/910
#
```

File: recipes/pylibmc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - python
+    - libmemcached
+    - glib
```
Comment (jakirkham) [inline recipes/pylibmc/meta.yaml]: Is this a run-time dependency too? Is there some reason it is only in `build`?

File: recipes/pylibmc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  run:
+    - python
+    - libmemcached
+
+test:
+  requires:
+    - python
+  imports:
+    - pylibmc
+    - _pylibmc
```
Comment (jakirkham) [inline recipes/pylibmc/meta.yaml]: What is the reason for importing this even though it has a `_` prefix? Is it an `.so` or something?

File: recipes/pylibmc/meta.yaml (hunk 3 above)
Comment (bollwyvl) [inline recipes/pylibmc/meta.yaml]: Nope, must have been left over from hacking...

File: recipes/pylibmc/build.sh (hunk 2 above)
Comment (bollwyvl) [inline recipes/pylibmc/build.sh]: Removing env shenanigans!

File: recipes/pylibmc/meta.yaml (hunk 4 above)
Comment (bollwyvl) [inline recipes/pylibmc/meta.yaml]: Yeah, it's a compiled module. Think I saw it somewhere else. I was trying to run the full test suite, but ran into trouble... will revisit at some point!

**Comment by bollwyvl** [comment]
Back to 💚, with no env prodding and glib removed. Hooray!

**Comment by bollwyvl** [comment]
Hm, did I skip windows incorrectly?

**Comment by pelson** [comment]
No, you're good. It is an old bug which we haven't got around to addressing. I should be able to get a chance to look at this today. Essentially, we can work around it by using conda-build-all on py34, so even if we don't get a fix, we will be able to get a workaround. xref https://github.com/conda-forge/conda-build-all-feedstock/issues/12.

**Comment by pelson** [comment]
Thanks @bollwyvl. Closing this one out. 👍

**Pull request merged**